tests/test_amcs.py

```diff
@@ -0,0 +1,1053 @@
+            expected_position=math.radians(6.0),
+            expected_velocity=-_MAX_SPEED,
+            expected_motion_state=mtdome.LlcMotionState.MOVING,
+        )
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 2.0,
+            expected_position=math.radians(2.0),
+            expected_velocity=-_MAX_SPEED,
+            expected_motion_state=mtdome.LlcMotionState.MOVING,
+        )
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 2.5,
+            expected_position=math.radians(0.0),
+            expected_velocity=crawl_velocity,
+            expected_motion_state=mtdome.LlcMotionState.CRAWLING,
+        )
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 4.0,
+            expected_position=math.radians(0.15),
+            expected_velocity=crawl_velocity,
+            expected_motion_state=mtdome.LlcMotionState.CRAWLING,
+        )
+
+    async def test_move_ten_zero_neg(self) -> None:
+        """Test the AmcsStatus when moving from position 10 to
+        position 0 degrees and then crawl in negative direction.
+        """
+        start_position = math.radians(10.0)
+        start_tai = START_TAI
+        target_position = 0.0
+        crawl_velocity = math.radians(-0.1)
+        expected_duration = START_MOTORS_ADD_DURATION + math.fabs(
+            (target_position - start_position) / _MAX_SPEED
+        )
+        await self.prepare_amcs(
+            start_position=start_position,
+            max_speed=_MAX_SPEED,
+            start_tai=start_tai,
+        )
+        duration = await self.amcs.moveAz(
+            position=target_position, velocity=crawl_velocity, start_tai=start_tai
+        )
+        assert expected_duration == pytest.approx(duration)
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 1.0,
+            expected_position=math.radians(6.0),
+            expected_velocity=-_MAX_SPEED,
+            expected_motion_state=mtdome.LlcMotionState.MOVING,
+        )
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 2.0,
+            expected_position=math.radians(2.0),
+            expected_velocity=-_MAX_SPEED,
+            expected_motion_state=mtdome.LlcMotionState.MOVING,
+        )
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 2.5,
+            expected_position=math.radians(0.0),
+            expected_velocity=crawl_velocity,
+            expected_motion_state=mtdome.LlcMotionState.CRAWLING,
+        )
+        await self.verify_amcs(
+            tai=START_TAI + START_MOTORS_ADD_DURATION + 4.0,
+            expected_position=math.radians(359.85),
+            expected_velocity=crawl_velocity,
+            expected_motion_state=mtdome.LlcMotionState.CRAWLING,
+        )
+
+    async def test_move_ten_threefifty_pos(self) -> None:
```

Comment:
I very strongly suggest writing a method `check_move(self, start_pos, end_pos, crawl_velocity)` that you can call with different arguments -- either in a loop or from different test_move_xxx methods. It would dramatically reduce the code duplication in this test file.

If it is too messy to be fully general (e.g. too much work to check short moves and long moves in the same method) then you could have different check methods for different categories of move.

Reply:
I agree. I actually started looking into that and then decided to revert the changes because it is a lot of work. I opened DM-35914 for this.